test/integration/default_elements.cc

```diff
@@ -0,0 +1,157 @@
+#include "sdf/parser.hh"
+#include "sdf/Frame.hh"
+#include "sdf/Model.hh"
+#include "sdf/Root.hh"
+#include "sdf/World.hh"
+#include "sdf/Filesystem.hh"
+#include "test_config.h"
+
+//////////////////////////////////////////////////
+TEST(ExplicitlySetInFile, EmptyRoadSphCoords)
+{
+  const std::string test_file =
+    sdf::filesystem::append(PROJECT_SOURCE_PATH, "test", "sdf",
+        "empty_road_sph_coords.sdf");
+
+  sdf::Root root;
+  sdf::Errors errors = root.Load(test_file);
+  EXPECT_TRUE(errors.empty());
+
+  sdf::ElementPtr element_ptr = root.Element();
+  EXPECT_TRUE(root.Element()->GetExplicitlySetInFile());
+
+  element_ptr = element_ptr->GetFirstElement();
+  EXPECT_TRUE(element_ptr->GetExplicitlySetInFile());
+
+  element_ptr = element_ptr->GetFirstElement();
+  EXPECT_TRUE(element_ptr->GetExplicitlySetInFile());
+
+  sdf::ElementPtr road_ptr = element_ptr->GetFirstElement();
+  EXPECT_FALSE(road_ptr->GetExplicitlySetInFile());
+
+  road_ptr = road_ptr->GetNextElement();
+  EXPECT_FALSE(road_ptr->GetExplicitlySetInFile());
+
+  element_ptr = element_ptr->GetNextElement();
+  EXPECT_TRUE(element_ptr->GetExplicitlySetInFile());
+
+  sdf::ElementPtr spherical_coords_ptr = element_ptr->GetFirstElement();
+  EXPECT_FALSE(spherical_coords_ptr->GetExplicitlySetInFile());
+
+  spherical_coords_ptr = element_ptr->GetNextElement();
+  EXPECT_FALSE(spherical_coords_ptr->GetExplicitlySetInFile());
+
+  spherical_coords_ptr = element_ptr->GetNextElement();
+  EXPECT_FALSE(spherical_coords_ptr->GetExplicitlySetInFile());
+
+  spherical_coords_ptr = element_ptr->GetNextElement();
+  EXPECT_FALSE(spherical_coords_ptr->GetExplicitlySetInFile());
+
+  spherical_coords_ptr = element_ptr->GetNextElement();
+  EXPECT_FALSE(spherical_coords_ptr->GetExplicitlySetInFile());
+
+  element_ptr = element_ptr->GetNextElement();
+  EXPECT_FALSE(element_ptr->GetExplicitlySetInFile());
+
+  element_ptr = element_ptr->GetNextElement();
+  EXPECT_FALSE(element_ptr->GetExplicitlySetInFile());
+
+  element_ptr = element_ptr->GetNextElement();
+  EXPECT_FALSE(element_ptr->GetExplicitlySetInFile());
+  
+  element_ptr = element_ptr->GetNextElement();
+  EXPECT_FALSE(element_ptr->GetExplicitlySetInFile());
+
+  sdf::ElementPtr physics_ptr = element_ptr->GetFirstElement();
+  EXPECT_FALSE(physics_ptr->GetExplicitlySetInFile());
+
+  physics_ptr = physics_ptr->GetNextElement();
+  EXPECT_FALSE(physics_ptr->GetExplicitlySetInFile());
+
+  physics_ptr = physics_ptr->GetNextElement();
+  EXPECT_FALSE(physics_ptr->GetExplicitlySetInFile());
+
+  element_ptr = element_ptr->GetNextElement();
+  EXPECT_FALSE(element_ptr->GetExplicitlySetInFile());
+
+  sdf::ElementPtr scene_ptr = element_ptr->GetFirstElement();
+  EXPECT_FALSE(scene_ptr->GetExplicitlySetInFile());
+
+  scene_ptr = scene_ptr->GetNextElement();
+  EXPECT_FALSE(scene_ptr->GetExplicitlySetInFile());
+
+  scene_ptr = scene_ptr->GetNextElement();
+  EXPECT_FALSE(scene_ptr->GetExplicitlySetInFile());
```

Comment:
Can this be refactored into a recursive function? Or is there a particular order of traversal of the elements that prevents generalization?

Reply:
I believe it can't be generalized as I don't see a trivial way to figure out weather a certain element is expected to be `ExplicitlySetInFile` or not